.circleci/osx_install_dependencies.sh

```diff
@@ -50,33 +50,65 @@ if [ ! -f /usr/local/lib/libz3.a ] # if this file does not exists (cache was not
 then
   brew update
   brew upgrade
-  brew install boost
   brew install cmake
   brew install wget
   brew install coreutils
   brew install diffutils
   brew install grep
-  ./scripts/install_obsolete_jsoncpp_1_7_4.sh
+
+  # writing to /usr/local/lib need administrative privileges.
+  sudo ./scripts/install_obsolete_jsoncpp_1_7_4.sh
+
+  # boost
+  boost_version="1.84.0"
+  boost_package="boost_${boost_version//./_}.tar.bz2"
+  boost_dir="boost_${boost_version//./_}"
+  wget "https://boostorg.jfrog.io/artifactory/main/release/$boost_version/source/$boost_package"
+  tar xf "$boost_package"
+  rm "$boost_package"
+  cd "$boost_dir"
+  ./bootstrap.sh --with-toolset=clang --with-libraries=thread,system,filesystem,program_options,serialization,test
+  sudo ./b2 -a address-model=64 architecture=arm+x86 install
```

Comment:
It will probably not check the `MAKEFLAGS`. Looking at the `b2` documenation: "Default number of jobs is the number of detected available CPU threads. " But maybe its saver just to specify it, how much threads should I define here?

Reply:
I would just leave it for now by that. I don't think it would create any problems not specifying a specific `-j` value.

Reply:
If `-j` just works, then you could use that.

But if not, use the same value as in `MAKEFLAGS`. It would be best not to hard-code it here though, because it may differ between jobs. Using `$MAKEFLAGS` directly is also risky because it could contain stuff other than just `-j`. You could add a new variable next to it in CI config and put just the number in it. Then use that variable here. It also should have a default in the script - probably `1` unless there's some value that makes it auto-detect.

Reply:
OK. Please add a comment about this though :P This wasn't obvious to us at all.